internal/controllers/reconciliation/controller.go

```diff
@@ -87,7 +87,7 @@ func (c *Controller) Reconcile(ctx context.Context, req *reconstitution.Request)
 	}
 	logger := logr.FromContextOrDiscard(ctx).WithValues("compositionGeneration", comp.Generation)
 
-	if comp.Status.CurrentSynthesis == nil {
+	if comp.Status.CurrentSynthesis == nil || comp.Status.CurrentSynthesis.Failed() {
```

Comment:
This is about 90% of the actual change being made in this PR 😆

Reply:
Okay so this just doesn't reconcile the contents of a failed synthesis.

Reply:
Right. We could add a third synthesis to the composition status to avoid blocking reconciliation, but I think terminal errors are rare enough this is fine